File: frontend/src/domain/entities/WafConfig.ts
// WAF configuration entity
export interface WafConfig {
  id: string;
  paranoiaLevel: number;
  ruleEngine: boolean;
  auditEngine: boolean;
  anomalyThreshold: number;
  blockedIps: string[];
  allowedIps: string[];
  customRules: CustomRule[];
  updatedAt: Date;
  updatedBy: string;
}

export interface CustomRule {
  id: string;
  name: string;
  pattern: string;
  action: RuleAction;
  enabled: boolean;
  priority: number;
}

export enum RuleAction {
  BLOCK = 'block',
  ALLOW = 'allow',
  LOG = 'log',
  REDIRECT = 'redirect'
}

export class WafConfigEntity implements WafConfig {
  constructor(
    public id: string,
    public paranoiaLevel: number,
    public ruleEngine: boolean,
    public auditEngine: boolean,
    public anomalyThreshold: number,
    public blockedIps: string[],
    public allowedIps: string[],
    public customRules: CustomRule[],
    public updatedAt: Date,
    public updatedBy: string
  ) {}

  isHighSecurity(): boolean {
    return this.paranoiaLevel >= 3;
  }

  isIpBlocked(ip: string): boolean {
    return this.blockedIps.includes(ip);
  }

  isIpAllowed(ip: string): boolean {
    return this.allowedIps.includes(ip);
  }

  getActiveRules(): CustomRule[] {
    return this.customRules.filter(rule => rule.enabled);
  }

  validate(): string[] {
    const errors: string[] = [];
    
    if (this.paranoiaLevel < 1 || this.paranoiaLevel > 4) {
      errors.push('Paranoia level must be between 1 and 4');
    }
    
    if (this.anomalyThreshold < 0) {
      errors.push('Anomaly threshold must be positive');
    }
    
    return errors;
  }
}

File: backend/app/domain/models/waf_config.py
# WAF configuration domain model
from sqlalchemy import Column, Integer, String, DateTime, Boolean, JSON
from sqlalchemy.ext.declarative import declarative_base
from datetime import datetime
from ...infrastructure.timezone import get_kst_now

Base = declarative_base()

class WafConfig(Base):
    __tablename__ = "waf_config"
    
    id = Column(Integer, primary_key=True)
    config_name = Column(String(100), unique=True, nullable=False)
    
    # Core settings
    paranoia_level = Column(Integer, default=1)
    rule_engine = Column(Boolean, default=True)
    audit_engine = Column(Boolean, default=True)
    anomaly_threshold = Column(Integer, default=5)
    
    # IP management
    blocked_ips = Column(JSON, default=list)
    allowed_ips = Column(JSON, default=list)
    
    # Custom rules
    custom_rules = Column(JSON, default=list)
    
    # Metadata
    updated_at = Column(DateTime, default=get_kst_now, onupdate=get_kst_now)
    updated_by = Column(String(100))
    
class CustomRule(Base):
    __tablename__ = "custom_rules"
    
    id = Column(Integer, primary_key=True)
    name = Column(String(200), nullable=False)
    pattern = Column(String(500), nullable=False)
    action = Column(String(20), nullable=False)  # block, allow, log, redirect
    enabled = Column(Boolean, default=True)
    priority = Column(Integer, default=100)
    
    created_at = Column(DateTime, default=get_kst_now)
    created_by = Column(String(100))